src/jit/regalloc.cpp

```diff
@@ -67,8 +65,84 @@ DWORD Compiler::getCanDoubleAlign()
     return DEFAULT_DOUBLE_ALIGN;
 #endif
 }
+
+//------------------------------------------------------------------------
+// shouldDoubleAlign: Determine whether to double-align the frame
+//
+// Arguments:
+//    refCntStk       - sum of     ref counts for all stack based variables
+//    refCntEBP       - sum of     ref counts for EBP enregistered variables
+//    refCntWtdEBP    - sum of wtd ref counts for EBP enregistered variables
+//    refCntStkParam  - sum of     ref counts for all stack based parameters
+//    refCntWtdStkDbl - sum of wtd ref counts for stack based doubles (including structs
+//                      with double fields).
+//
+// Return Value:
+//    Returns true if this method estimates that a double-aligned frame would be beneficial
+//
+// Notes:
+//    The impact of a double-aligned frame is computed as follows:
+//    - We save a byte of code for each parameter reference (they are frame-pointer relative)
+//    - We pay a byte of code for each non-parameter stack reference.
+//    - We save the misalignment penalty and possible cache-line crossing penalty.
+//      This is estimated as 0 for SMALL_CODE, 16 for FAST_CODE and 4 otherwise.
+//    - We pay 7 extra bytes for:
+//        MOV EBP,ESP,
+//        LEA ESP,[EBP-offset]
+//        AND ESP,-8 to double align ESP
+//    - We pay one extra memory reference for each variable that could have been enregistered in EBP (refCntWtdEBP).
+//
+//    If the misalignment penalty is estimated to be less than the bytes used, we don't double align.
+//    Otherwise, we compare the weighted ref count of ebp-enregistered variables aginst double the
+//    ref count for double-aligned values.
+//
+bool Compiler::shouldDoubleAlign(
+    unsigned refCntStk, unsigned refCntEBP, unsigned refCntWtdEBP, unsigned refCntStkParam, unsigned refCntWtdStkDbl)
```

Comment:
Yes, that would be an improvement that I would like to see.
Though this particular method isn't exactly "common" to all architectures, as it only applies to x86.